dcpy/connectors/edm/publishing.py

```diff
@@ -104,11 +108,20 @@ def previous(product: str, version: str | versions.Version) -> versions.Version:
         if version_obj > latest:
             return latest
         else:
-            raise Exception(
+            raise LookupError(
                 f"{product} - {version} is not published and appears to be 'older' than latest published version. Cannot determine previous."
             )
 
 
+def try_get_previous_version(
+    product: str, version: str | versions.Version
+) -> versions.Version | None:
+    try:
+        return get_previous_version(product, version)
+    except (LookupError, ValueError):  # versions not found, or don't parse correctly
```

Comment:
Should we let these Exceptions bubble up? Or is there a legitimate case where there wouldn't be a previous build?

Reply:
ValueError potentially. I mainly didn't want to break anybody's code right now, but it's going to be tough to sort out issues if we don't raise errors.

The LookupErrors we have some "valid" cases - new data product, or repeat of original build where there truly is no previous. I'd rather return None and let build logic decide what to do than throw an error here, personally

Reply:
Gotcha with the LookupError. I'd be a little worried that catching `ValueError`s might cause some silent issues, but totally get erring on the side of not breaking things.